### PR TITLE
Recommendations for hardware size and shard count

### LINKED ISSUE #454: Choosing shard count
Currently we have https://docs.citusdata.com/en/v6.2/faq/faq.html#how-do-i-change-the-shard-count-for-a-hash-partitioned-table . This is more relevant to real-time analytics (shard on entity_id) workloads.
But for Multi-tenant SAAS workloads, where you shard on tenant_id it changes. We recommend anywhere from 32 to 128 shards. For smaller workloads say <100GB, you could start with 32 shards and for larger workloads you could choose 64 or 128. This means that you have the leeway to scale from 32 to 128 worker machines.

**Where to put on docs:**
A sub section in sizing (or)
Update the existing FAQ.

### PR DESCRIPTION
**Pull request by begriffs**
Fixes #454

### CODE REVIEW

**Comment by begriffs** [comment]
Included a commit which fixes #453 since this PR already added a sizing section.

**Comment by begriffs** [comment]
This PR updates the FAQ entry about changing shard count. It essentially leaves the question unanswered and talks about good shard counts for new tables. I could also incorporate content from https://github.com/citusdata/citus_docs/pull/450 about `citus.shard_count`.

**Comment by sumedhpathak** [comment]
Do we modify this FAQ in any way?: https://docs.citusdata.com/en/v7.0/faq/faq.html#how-do-i-choose-the-shard-count-when-i-hash-partition-my-data? Maybe we add a brief blurb and point it to this new section?

**Comment by sumedhpathak** [comment]
Also, let's incorporate the change from #450 here, as it probably makes sense to give a more pointed answer there.

File: admin_guide/production_sizing.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
+.. _production_sizing:
+
+Production Sizing
```
Comment (sumedhpathak) [inline admin_guide/production_sizing.rst]: Now that we have our fancy new front-page, could we add this to the migration button page?

File: admin_guide/production_sizing.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+$$$$$$$$$$$$$$$$$
+
+This section explores configuration settings for running a cluster in production.
+
+.. _prod_shard_count:
+
+Shard Count
+===========
+
+The number of nodes in a cluster is easy to change (see :ref:`scaling_out_cluster`), but the number of shards to distribute among those nodes is more difficult to change after cluster creation. Choosing the shard count is a balance between the flexibility of having more shards, and the overhead for query planning and execution across them.
```
Comment (sumedhpathak) [inline admin_guide/production_sizing.rst]: Could we clarify that number of shards is for an existing distributed table? (not as a whole cluster-wide)?

File: admin_guide/production_sizing.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+The optimal choice varies depending on your access patterns for the data. For instance, in the :ref:`mt_blurb` use-case we recommend choosing between **32 - 128 shards**.  For smaller workloads say <100GB, you could start with 32 shards and for larger workloads you could choose 64 or 128. This means that you have the leeway to scale from 32 to 128 worker machines.
```
Comment (sumedhpathak) [inline admin_guide/production_sizing.rst]: The real-time analytics answer makes sense to me, but I don't have a clear understanding for this one. We say "you could start with 32 shards", which implies this can be changed later. So we should change that.
Also, I don't think anyone would not choose 64 or 128 based on this recommendation? We don't really say why you would want to choose 32 or 64 or 128. @saicitus/ @samay-sharma  thoughts?

File: admin_guide/production_sizing.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+--------------------------
+
+For those migrating to Citus from an existing single-node database instance, we recommend choosing a cluster where the number of worker cores and RAM in total equals that of the original instance. In such scenarios we have seen 2-3x performance improvements because sharding improves resource utilization, allowing smaller indices etc.
+
+For the coordinator node, choose a compute-optimized machine. The coordinator is responsible for the computationally intensive job of routing queries to the right workers. The number of cores needed on the coordinator depends on your existing workload (write/read throughput). By default in Citus Cloud the workers use Amazon EC2 instance type R4S, and the coordinator uses C4S.
+
+Real-Time Analytics Use-Case
+----------------------------
+
+**Total cores:** when working data fits in RAM, you can expect a linear performance improvement on Citus proportional to the number of worker cores. To determine the right number of cores for your needs, consider the current latency for queries in your single-node database and the required latency in Citus. Divide current latency by desired latency, and round the result.
```
Comment (sumedhpathak) [inline admin_guide/production_sizing.rst]: In practice this isn't as clear-cut, as postgreSQL contention often limits us. @saicitus @samay-sharma Are we OK making such a recommendation?

File: admin_guide/production_sizing.rst [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+=====================
+
+The size of a cluster, in terms of number of nodes and their hardware capacity, is easy to change. (:ref:`Scaling <cloud_scaling>` on Citus Cloud is especially easy.) However you still need to choose an initial size for a new cluster. Here are some tips for a reasonable initial cluster size.
+
+Multi-Tenant SaaS Use-Case
+--------------------------
+
+For those migrating to Citus from an existing single-node database instance, we recommend choosing a cluster where the number of worker cores and RAM in total equals that of the original instance. In such scenarios we have seen 2-3x performance improvements because sharding improves resource utilization, allowing smaller indices etc.
+
+For the coordinator node, choose a compute-optimized machine. The coordinator is responsible for the computationally intensive job of routing queries to the right workers. The number of cores needed on the coordinator depends on your existing workload (write/read throughput). By default in Citus Cloud the workers use Amazon EC2 instance type R4S, and the coordinator uses C4S.
```
Comment (sumedhpathak) [inline admin_guide/production_sizing.rst]: "The coordinator is responsible for the computationally intensive job of routing queries" --> We typically say the co-ordinator doesn't do as-much work, and we don't deem query routing as computationally intensive (from a communicatino standpoint). Can we reword this to say something like the co-ordinator doesn't hold any data (and only routes and does final aggregation), and so we can use compute-optimized machines, rather than memory-optimized?

File: admin_guide/production_sizing.rst (hunk 1 above)
Comment (begriffs) [reply admin_guide/production_sizing.rst]: (Added)

**Comment by begriffs** [comment]
Addressed the comments so far, and awaiting feedback from @samay-sharma or @saicitus

File: faq/faq.rst [hunk 6]
```diff
@@ -14,13 +14,6 @@ How do I add nodes to an existing Citus cluster?
 
 You can add nodes to a Citus cluster by calling the master_add_node UDF with the hostname (or IP address) and port number of the new node. After adding a node to an existing cluster, it will not contain any data (shards). Citus will start assigning any newly created shards to this node. To rebalance existing shards from the older nodes to the new node, the Citus Enterprise edition provides a shard rebalancer utility. You can find more information about shard rebalancing in the :ref:`cluster_management` section.
 
-How do I change the shard count for a hash partitioned table?
```
Comment (sumedhpathak) [inline faq/faq.rst]: I'd prefer retaining this question, as it did come up somewhat frequently, rather than adding a note to the end of the other question.

File: admin_guide/production_sizing.rst [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+This section explores configuration settings for running a cluster in production.
+
+.. _prod_shard_count:
+
+Shard Count
+===========
+
+The number of nodes in a cluster is easy to change (see :ref:`scaling_out_cluster`), but the number of shards to distribute among those nodes is more difficult to change after cluster creation. Choosing the shard count for each distributed table is a balance between the flexibility of having more shards, and the overhead for query planning and execution across them.
+
+The optimal choice varies depending on your access patterns for the data. For instance, in the :ref:`mt_blurb` use-case we recommend choosing between **32 - 128 shards**.  For smaller workloads say <100GB, you could start with 32 shards and for larger workloads you could choose 64 or 128. This means that you have the leeway to scale from 32 to 128 worker machines.
```
Comment (sumedhpathak) [inline admin_guide/production_sizing.rst]: Should we create a separate sub-section for multi-tenant vs real-time like we did below for hardware sizing? (For consistency). Or do you think its too sparse for that?

File: faq/faq.rst [hunk 8]
```diff
@@ -59,13 +52,29 @@ Other queries which, by contrast, combine data from multiple nodes, do not suppo
...
+One of the choices when first distributing a table is its shard count. This setting can be set differently for each table, and the optimal value depends on use-case. It is possible, but difficult, to change the count after cluster creation, so use these guidelines to choose the right size.
+
+In the :ref:`mt_blurb` use-case we recommend choosing between 32 - 128 shards.  For smaller workloads say <100GB, you could start with 32 shards and for larger workloads you could choose 64 or 128. This means that you have the leeway to scale from 32 to 128 worker machines.
+
+In the :ref:`rt_blurb` use-case, shard count should be related to the total number of cores on the workers. To ensure maximum parallelism, you should create enough shards on each node such that there is at least one shard per CPU core. We typically recommend creating a high number of initial shards, e.g. 2x or 4x the number of current CPU cores. This allows for future scaling if you add more workers and CPU cores.
```
Comment (sumedhpathak) [inline faq/faq.rst]: This info is good, but could you also link to the production sizing section (e.g. for more information see ...)?

File: faq/faq.rst [hunk 9]
```diff
@@ -59,13 +52,29 @@ Other queries which, by contrast, combine data from multiple nodes, do not suppo
...
+
+  SET citus.shard_count = 64;
+  -- any tables distributed at this point will have
+  -- sixty-four shards
+
+How do I change the shard count for a hash partitioned table?
+-------------------------------------------------------------
 
-We typically recommend creating a high number of initial shards, e.g. 2x or 4x the number of current CPU cores. This allows for future scaling if you add more workers and CPU cores.
+Note that it is not straightforward to change the shard count of an already distributed table. If you need to do so, please `Contact Us <https://www.citusdata.com/about/contact_us>`_. It's good to think about shard count carefully at distribution time, see :ref:`faq_choose_shard_count`.
```
Comment (sumedhpathak) [inline faq/faq.rst]: I'm good to retain this note, or link to the other FAQ item, but I'd prefer to keep the other FAQ as well.

File: faq/faq.rst (hunk 6 above)
Comment (begriffs) [reply faq/faq.rst]: The github diff layout doesn't make it very apparent, but this question still exists in the FAQ. It's just moved below "How do I choose the shard count." Still a top-level entry in the document, though its answer is pretty brief.

File: faq/faq.rst (hunk 9 above)
Comment (begriffs) [reply faq/faq.rst]: It is actually the same question, but moved lower. Are you saying you prefer the old answer to the question over this new text?

File: faq/faq.rst (hunk 8 above)
Comment (begriffs) [reply faq/faq.rst]: Good idea. Done.

File: admin_guide/production_sizing.rst (hunk 7 above)
Comment (begriffs) [reply admin_guide/production_sizing.rst]: Sure, we do it that way in the next section so it's more consistent to do it here too.
